test/PositionToken.ts

```diff
@@ -1,15 +1,116 @@
+import { SignerWithAddress } from "@nomiclabs/hardhat-ethers/signers";
 import { expect } from "chai";
+import { ContractFactory } from "ethers";
 import { ethers } from "hardhat";
+import { PositionToken } from "../types/contracts";
+import { buildContractTestContext, ContractTestContext } from "./ContractTestContext";
 
-describe("PositionToken test suit", async function () {
-    let positionToken;
+describe("PositionToken test suit", function () {
+    let r: ContractTestContext;
+    let ptContract: ContractFactory;
 
-    beforeEach(async () => {
-        const ptContract = await ethers.getContractFactory("PositionToken");
-        positionToken = await ptContract.deploy();
+    describe("Roles and initialization", async function () {
+        before(async () => {
+            r = await buildContractTestContext();
+            ptContract = await ethers.getContractFactory("PositionToken");
+        });
+
+        it("Should be built properly by ContractTestContextt", async function () {
+            expect(r.positionToken).to.not.be.undefined;
+        });
+
+        it("Should not allow non admin to init", async function () {
+            const addr1PositionToken = await ptContract.deploy(r.addr1.address) as PositionToken;
+            const initPromise = addr1PositionToken.init(r.addr2.address);
+            await expect(initPromise).to.be.revertedWith("onlyAdmin: Caller is not admin");
+        });
+
+        it("Should not allow non executive to mint", async function () {
+            await r.positionToken.init(r.addr1.address);
+            const mintPromise = r.positionToken.safeMint(r.addr1.address);
+            await expect(mintPromise).to.be.revertedWith("onlyExecutive: Caller is not executive");
+        });
     });
 
-    it("Should exist", async function () {
-        expect(positionToken).to.not.be.undefined;
+    describe("After initialized with owner set to executive", async function () {
+        const firstTokenId = 0;
+        const secondTokenId = 1;
+        const thirdTokenId = 2;
+        let firstTokenOwner: SignerWithAddress;
+        let firstTokenOwnerAddress: string;
+        let secondTokenOwner: SignerWithAddress;
+        let secondTokenOwnerAddress: string;
+        let thirdTokenOwner: SignerWithAddress;
+        let thirdTokenOwnerAddress: string;
+
+        before(async function () {
+            r = await buildContractTestContext();
+            await r.positionToken.init(r.owner.address);
```

Comment:
positionToken.init should happen inside buildContractTestContext like the other contracts. I suggest modifying describe "Roles and initialization" instead of making the init process for  positionToken unique

Reply:
Yeah this is not quite right because its going to initialize twice anyways (once in ContractTestContext and once here). So I need to fix that. Basically my thinking here was I want to test PositionToken here completely independently of leverageEngine (which is what ContractTestContext sets to executive). However I can just use `connect` to simulate that I think and do what you are suggesting.

Reply:
How does it help this test that you are using r.owner.address instead of levEng?

Reply:
Because r.owner is the user that is running the contracts by default. So for the other commands that don't use `connect` at all they are running as r.owner

Reply:
I see, basically you want to have admin/executive access and you cant really connect as levEng. Once we add access control to all contracts its going to be wide spread issue. 

I suggest adding an optional param to `buildContractTestContext` which can take a fake admin/executive and init all contracts with that

Reply:
So I forgot that there doesn't really seem to be a way to connect as a contract separately from the contract itself. I can't do `r.positionToken.connect(r.leverageEngine)`. I did find `r.leverageEngine.signer` is a thing but it has a different address than the contract apparently so I'm not sure what it's for exactly.